core/bones/credential.py

```diff
@@ -33,8 +35,9 @@ def unserialize(self, valuesCache, name):
         """
         return {}
 
-    def singleValueFromClient(self, value, skel, name, origData):
-        err = self.isInvalid(value)
-        if not err:
+    def singleValueFromClient(self, value: Any, skel: 'SkeletonInstance',
+                              bone_name: str, client_data: dict
+                              ) -> tuple[Any, list[ReadFromClientError] | None]:
```

Comment:
Is it necessary to provide full type annotations in sub-classes as well?
I would like to keep maintenance overhead low, if possible.